petastorm/unischema.py

```diff
@@ -221,7 +271,24 @@ def dict_to_spark_row(unischema, row_dict):
                 raise ValueError('Field {} is not "nullable", but got passes a None value')
         encoded_dict[field_name] = schema_field.codec.encode(schema_field, value) if value is not None else None
 
-    return Row(**encoded_dict)
+    if not unischema.column_bundles:
+        return Row(**encoded_dict)
+    else:
+        bundled_dict = {}
+        included_field_names = set()
+        for bundle_name, bundle_columns in unischema.column_bundles.items():
```

Comment:
Consider making bundling into a separate function. Maybe separate module/class? The serialization part is interesting. Maybe it is as fast as pyarrow.serialize, but without the limitations? Would be nice if the piece of code that allows that would be easily accessible (an equivalent of https://github.com/uber/petastorm/pull/185/commits/2ee23f8891a889269a0f2da872628cdef908923b)